### ISSUE TITLE
ESLint

### ISSUE THREAD

**Issue by jonathandewitt-dev** [Thunder-Solutions/Thunder-State#23]
It turns out I never added eslint to this project. I normally would have used standardjs, down to the no-semicolons. These days I'm developing more of a taste for semicolons again, so I'd probably flip that rule at least. Trailing commas are great. Might consider some rules around avoiding mutations? Prefer const, stuff like that. I like one empty line above comments to make it readable. I dunno, open to suggestions here, doesn't really matter so much, as long as we have some uniform lint checks

**Comment by bcdewitt** [comment]
`eslint-plugin-functional` proved to be nice for me, with some tweaks (especially needed exceptions for testing)

**Comment by jonathandewitt-dev** [comment]
I was thinking about this one the other day, yeah. I'm not sure what all rules that plugin includes, but maybe worth calling out that sometimes an impure function or mutation is necessary here and there, might be a pain if it _always_ fails the linter. I'm sure this is what you mean by some tweaks, lol.

**Comment by bcdewitt** [comment]
Yeah, pretty much what I meant. Mutations are obviously needed in a state management library, but you likely want them put in a single location - in a folder where you can relax the rules and do a different style of testing, for example

**Comment by bcdewitt** [comment]
Testing libraries are also heavily based on throwing exceptions rather than returning pass/fail result objects. The latter is expected by the lint rules. At least, the stricter set I was using checked for return values
